tests/components/bryant_evolution/test_climate.py

```diff
@@ -0,0 +1,399 @@
+        self._allow_reads_cond = asyncio.Condition()
+        self._is_active = False
+
+    async def read_current_temperature(self) -> int | None:
+        async with self._allow_reads_cond:
+            await self._allow_reads_cond.wait_for(self._are_reads_allowed)
+            return self._temp
+
+    async def read_cooling_setpoint(self) -> int | None:
+        async with self._allow_reads_cond:
+            await self._allow_reads_cond.wait_for(self._are_reads_allowed)
+            return self._clsp
+
+    async def read_heating_setpoint(self) -> int | None:
+        async with self._allow_reads_cond:
+            await self._allow_reads_cond.wait_for(self._are_reads_allowed)
+            return self._htsp
+
+    def _are_reads_allowed(self) -> bool:
+        return self._allow_reads
+
+    async def read_fan_mode(self) -> str | None:
+        async with self._allow_reads_cond:
+            await self._allow_reads_cond.wait_for(self._are_reads_allowed)
+            return self._fan
+
+    async def set_cooling_setpoint(self, temperature: int) -> bool:
+        async with self._allow_reads_cond:
+            self._clsp = temperature
+            return True
+
+    async def set_heating_setpoint(self, temperature: int) -> bool:
+        async with self._allow_reads_cond:
+            self._htsp = temperature
+            return True
+
+    async def read_hvac_mode(self) -> tuple[str, bool] | None:
+        async with self._allow_reads_cond:
+            await self._allow_reads_cond.wait_for(self._are_reads_allowed)
+            if not self._mode:
+                return None
+            return (self._mode, self._is_active)
+
+    async def set_fan_mode(self, fan_mode: str) -> bool:
+        async with self._allow_reads_cond:
+            self._fan = fan_mode
+            return True
+
+    async def set_allow_reads(self, allow: bool) -> None:
+        async with self._allow_reads_cond:
+            self._allow_reads = allow
+            self._allow_reads_cond.notify_all()
+
+    async def set_hvac_mode(self, hvac_mode: HVACMode) -> bool:
+        async with self._allow_reads_cond:
+            self._mode = hvac_mode
+            return True
+
+    async def set_is_active(self, is_active: bool) -> None:
+        async with self._allow_reads_cond:
+            self._is_active = is_active
+
+
+@pytest.fixture
+async def mock_evolution_entry(
+    hass: HomeAssistant,
+) -> MockConfigEntry:
+    """Configure and return a Bryant evolution integration."""
+    hass.config.units = US_CUSTOMARY_SYSTEM
+    with patch(
+        "evolutionhttp.BryantEvolutionLocalClient.get_client",
+        return_value=_FakeEvolutionClient(),
+    ):
+        entry = MockConfigEntry(
+            domain=DOMAIN,
+            data={CONF_FILENAME: "/dev/ttyUSB0", CONF_SYSTEM_ZONE: [(1, 1)]},
+        )
+        entry.add_to_hass(hass)
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+        return entry
```

Comment:
This is something for `conftest.py`.
For inspiration on what I personally think are good and nice tests, checkout `airgradient`